Update all features and plugins for 2025-09 release cycle

ptziegler opened the pull request.

ptziegler commented on pom.xml:
```diff
@@ -55,6 +55,8 @@
 				<artifactId>tycho-packaging-plugin</artifactId>
 				<version>${tycho-version}</version>
 				<configuration>
+					<timestampProvider>jgit</timestampProvider>
```
That is the final piece responsible for those dummy baseline changes. With this, the date of the last commit is used as the qualifier, as opposed to the time-of-build.

azoitl commented on pom.xml:
Do we need to update the repo in line 361 also, or can this profile be removed?

ptziegler replied:
You're right. I think I messed up the line numbers. The "baseline" profile checks against the minimum version. The "master" profile is a placeholder because the profile specified in the yaml file needs to exist. Otherwise a warning is generated.

https://github.com/eclipse-gef/gef-classic/blob/52e81e7172431bebd2fc79243131f2dbe084136e/.github/workflows/maven.yml#L37

The "master" profile always points to `target-platform/GEF_classic.target`, while the "baseline" profile points to `gef-classic-3.20.0.target`. I don't think either need to be changed.

azoitl reviewed (commented):
Ah you where faster, I had that prepared when doing the release, but then never found the time to complete it. 

Should we also add the changelog.md update here?
Tycho has a new release, would be a nice chance to update that as part of this PR or?

ptziegler commented:
> Should we also add the changelog.md update here?

Added.

>  Tycho has a new release, would be a nice chance to update that as part of this PR or?

Also added.

ptziegler commented on org.eclipse.zest.cloudio/src/org/eclipse/zest/cloudio/TagCloud.java:
```diff
@@ -628,7 +628,8 @@ protected int layoutWords(Collection<Word> wordsToUse, IProgressMonitor monitor)
 		gc.dispose();
-		selectionLayerImage = new Image(getDisplay(), textLayerImage.getBounds());
+		Rectangle textLayerBounds = textLayerImage.getBounds();
+		selectionLayerImage = new Image(getDisplay(), textLayerBounds.width, textLayerBounds.height);
```
The constructor Image(Display, Rectangle) has been deprecated with 2025-06.

ptziegler commented on org.eclipse.gef/src/org/eclipse/gef/ui/palette/FlyoutPaletteComposite.java:
```diff
@@ -1581,6 +1581,7 @@ private static class DragCursors {
 		 * @param code the code
 		 * @return the cursor
 		 */
+		@SuppressWarnings("removal") // Still required for 2024-06
```
Weirdly enough, the original icons no longer show up anymore?

azoitl replied:
I'm not sure but I thought I read somewhere that with pngs and maybe also SVGs you can create cursors without mask. Have you tried that? Or I'm totaly on the wrong page?

ptziegler replied:
Yes, of course you can. And that's why e.g. `IMG_OBJS_DND_LEFT_SOURCE` and `IMG_OBJS_DND_LEFT_MASK` have been deprecated in favor of `IMG_OBJS_DND_LEFT`. 

But a) this new constant only exists since 2024-09, which is why I brought up raising the minimum Eclipse version and b) the current handling breaks existing functionality.

azoitl replied:
Ah stupid me. I now remember the discussion with had. Should we merge then and decide during M1 how to proceed?

ptziegler replied:
Sounds good to me. The issue already exists with the 2025-06 and I think it's better to handle this separately from the 2025-09 update.
